cpp/convert_rss_db_tables.cc

```diff
@@ -0,0 +1,116 @@
+        *feed_info = url_and_feed_info->second;
+        return true;
+    }
+
+    vufind_connection->queryOrDie("SELECT id,subsystem_types FROM tuefind_rss_feeds WHERE feed_url="
+                                  + vufind_connection->escapeAndQuoteString(feed_url));
+    auto result_set(vufind_connection->getLastResultSet());
+    if (result_set.empty()) {
+        static std::unordered_set<std::string> known_missing;
+        if (known_missing.find(feed_url) == known_missing.end()) {
+            LOG_WARNING("found no tuefind_rss_feeds.id for \"" + feed_url + "\"!");
+            known_missing.emplace(feed_url);
+        }
+        return false;
+    }
+
+    const auto row(result_set.getNextRow());
+    std::vector<std::string> subsystem_types;
+    StringUtil::Split(row["subsystem_types"], ',', &subsystem_types);
+    *feed_info = FeedInfo(row["id"], StlHelpers::VectorToSet(subsystem_types));
+    urls_to_feed_infos_map[feed_url] = *feed_info;
+    return true;
+}
+
+
+void CopyItem(DbConnection * const db_writer, const std::string &feed_id, const std::string &item_id,
+              const std::string &item_url, const std::string &item_title, const std::string &item_description,
+              const std::string &pub_date, const std::string &insertion_time)
+{
+    db_writer->queryOrDie("INSERT INTO tuefind_rss_items SET rss_feeds_id=" + feed_id + ",item_id='" + item_id
+                          + "',item_url=" + db_writer->escapeAndQuoteString(item_url) + ",item_title="
+                          + db_writer->escapeAndQuoteString(item_title) + ",item_description="
+                          + db_writer->escapeAndQuoteString(item_description) + ",pub_date='"
+                          + pub_date + "',insertion_time='" + insertion_time + "'");
+}
+
+
+} // unnamed namespace
+
+
+int Main(int /*argc*/, char */*argv*/[]) {
+    auto db_reader((DbConnection()));
+    auto db_writer(VuFind::GetDbConnection());
+
+    db_reader.queryOrDie("SELECT * FROM rss_aggregator");
+    auto result_set(db_reader.getLastResultSet());
+    while (const auto row = result_set.getNextRow()) {
+        FeedInfo feed_info;
+        if (not GetRSSFeedsID(db_writer.get(), row["feed_url"], &feed_info))
+            continue;
+        if (unlikely(not feed_info.isCompatibleWith(row["flavour"])))
+            LOG_ERROR("Item w/ item_id \"" + row["item_id"] + " has a flavour \"" + row["flavour"] +
+                      "\" which is incompatible with the subsystem_types \""
+                      + StlHelpers::ContainerToString(feed_info.subsystem_types_.cbegin(),
+                                                      feed_info.subsystem_types_.cend(), ","));
+        CopyItem(db_writer.get(), feed_info.id_, row["item_id"], row["item_url"], row["item_title"],
+                 row["item_description"], row["pub_date"], row["insertion_time"]);
+    }
+    db_reader.queryOrDie("DROP TABLE IF EXISTS vufind.rss_aggregator");
```

Comment:
I guess this line is just for testing purposes and should be removed later

Reply:
.... and: This is executed on the db_reader (ub_tools database) so it should never have any effect since the reader has the connection to the ub_tools and not the vufind database.

Reply:
Good catch.  It should have been ub_tools.rss_aggregator.  In any case there is no table named rss_aggregator in the vufind database.